airflow/providers/amazon/aws/operators/sagemaker_base.py

```diff
@@ -41,12 +43,11 @@ class SageMakerBaseOperator(BaseOperator):
     integer_fields = []  # type: Iterable[Iterable[str]]
 
     @apply_defaults
-    def __init__(self, *, config, aws_conn_id='aws_default', **kwargs):
+    def __init__(self, *, config: dict, aws_conn_id: str = 'aws_default', **kwargs):
         super().__init__(**kwargs)
 
         self.aws_conn_id = aws_conn_id
         self.config = config
-        self.hook = None
```

Comment:
This attribute is directly referenced from child classes. To avoid `None` I defined it as cached_property.

Reply:
Well, I liked this one :)